### PR TITLE
AB#88304 added question 6 to section 106, added integration test for sites

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            continueButton,
+            (nameof(SiteModel.Section106GeneralAgreement), "true"));
+
+        // then
+        section106AffordableHousingPage
+            .UrlEndWith(SitePagesUrl.SiteSection106AffordableHousing(SiteData.SiteId))
+            .HasTitle(SitePageTitles.SiteSection106AffordableHousing);
+
+        SaveCurrentPage();
+    }
```
Comment (jjankowski87) [inline src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs:101]: Tests 4 - 8 can be replaced with `TestQuestionPage` from base class, something like
```
await TestQuestionPage(
    SitePagesUrl.SiteSection106GeneralAgreement(SiteData.SiteId),
    SitePageTitles.SiteSection106Agreement,
    SitePagesUrl.SiteSection106AffordableHousing(SiteData.SiteId),
    (nameof(SiteModel.Section106GeneralAgreement), "true"));
```

File: src/HE.Investments.Account.WWW/Properties/launchSettings.json [hunk 2]
```diff
@@ -15,13 +15,22 @@
       "environmentVariables": {
         "ASPNETCORE_ENVIRONMENT": "Development"
       }
+    },
```
Comment (jjankowski87) [inline src/HE.Investments.Account.WWW/Properties/launchSettings.json:18]: revert

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs [hunk 3]
```diff
@@ -47,7 +47,7 @@ public Section106()
 
     public bool? CapitalFundingEligibility { get; }
 
-    public string? ConfirmationFromLocalAuthority { get; }
+    public string? LocalAuthorityConfirmation { get; }
 
     public bool IsAnswered()
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs:52]: Please implement the method and add unit tests for it.

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs [hunk 4]
```diff
@@ -59,13 +59,28 @@ public bool IsAnswered()
         return false;
     }
 
+    public bool IsIneligibleDueToAffordableHousing()
+    {
+        return AffordableHousing == true && OnlyAffordableHousing == false && AdditionalAffordableHousing == false;
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs:64]: `== true` is redundant.

File: src/HE.Investments.AHP.IntegrationTests/Framework/AhpIntegrationTest.cs [hunk 5]
```diff
@@ -21,11 +22,22 @@ protected AhpIntegrationTest(IntegrationTestFixture<Program> fixture)
...
         ApplicationData = applicationData;
+
+        var siteData = GetSharedDataOrNull<SiteData>(nameof(SiteData));
+        if (siteData is null)
+        {
+            siteData = new SiteData();
+            SetSharedData(nameof(SiteData), siteData);
+        }
+
+        SiteData = siteData;
```
Comment (Carq) [inline src/HE.Investments.AHP.IntegrationTests/Framework/AhpIntegrationTest.cs:33]: Please wrap the code into well-named method. The same for code above (ApplicationData)

**Review by Carq — APPROVED** [review]

File: src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs:101]: yes, I noticed that method after commit, will do that.

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs (hunk 3 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs:52]: done

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs (hunk 4 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs:85]: lets catch and talk about it

File: src/HE.Investments.AHP.IntegrationTests/Framework/AhpIntegrationTest.cs (hunk 5 above)
Comment (rafalgajewski) [reply src/HE.Investments.AHP.IntegrationTests/Framework/AhpIntegrationTest.cs]: done

**Pull request merged**